BEST_PRACTICE.md

```diff
@@ -0,0 +1,247 @@
+then do this: `"install": "./configure && make && make install"`). `"uninstall"` is the command
+to uninstall your project, [more on that later](#install-and-uninstall-executables).
+
+***NOTE:** Make sure you have a relese as the same version in your `package.json` file, otherwise
+the download will fail. If you allwasy want your package at the latest version, then put `master`
+as your version.*
+
+## Making your own libraries
+
+Now that your know how to use libraries, heres how to make your own:
+
+Like before, heres a typical project directory tree:
+
+```
+your-library-c/
+│
+├── deps/
+│   ├── path-join.c/
+│   │   ├── path-join.h
+│   │   ├── path-join.c
+│   │   └── package.json
+│   │
+│   └── strdup/
+│       ├─ strdup.h
+│       ├─ strdup.c
+│       └─ package.json
+│
+├── LICENSE
+│
+├── Makefile
+│
+├── README.md
+│
+├── package.json
+│
+├── src/
+│   ├── library.c
+│   ├── library.h
+│   └── etc...
+│
+└── test.sh
+```
+
+Also like before, your have a `deps` directory (depending on your library, you may not need any
+dependencies). Your `Makefile` in this case it is only for the `test.sh`, not needed for installing.
+`package.json` contains your library name, dependencies (if you need them), keywords, etc... In
+`src/` contains your make code (usally the same name as your library). And you have your `test.sh`
+used for testing.
+
+### Example package.json for libraries
+
+```
+{
+  "name": "your-lib-name",
+  "version": "1.0.0",
+  "repo": "your-github-name/library-name",
+  "description": "What my library does",
+  "keywords": [
+    "somthing",
+    "cool",
+    "mylib",
+  ],
+  "dependencies": {
+    "stephenmathieson/path-join.c": "0.0.6",
+    "clibs/strdup": "*"
+  }
+  "license": "YOUR_LIB LICENSE",
+  "src": [
+    "src/library.c",
+    "src/library.h"
+  ],
+}
+```
+
+The main difrances (between this, and the executable `package.json`), is now there is `"src"`,
+this is where your make library source code is, your can change it, but src is pritty standard.
+
+**TIP:** In the `"dependencies"` section, if you define `"*"` as the version, clib will install
+the latest version of that library.
+
+***NOTE:** Just like your executable package, you will want to tag a relese with the same name
+as your version specified in your `package.json`.*
+
+## Install and uninstall executables packages
+
+Installing executables is best done as root (with `sudo`), here is a typical install command:
+
+```bash
+$ sudo clib install visionmedia/mon
+```
+
+To uninstall a package, (as of today) your need to install `clib-uninstall`:
+
+```bash
+$ sudo clib install clib-uninstall
+```
+
+***TIP:** If you don't specify a username when installing a package or library, clib will
+download that package in the default location: `https://github.com/clibs/`.*
+
+After you install `clib-uninstall` you can use it like so:
+
+```bash
+$ sudo clib-uninstall <username/package-name>
+
+# for example:
+$ sudo clib-uninstall visionmedia/mon
+```
+
+<br>
```

Comment:
What is this for?

Reply:
Because I'm weird. Just a break at the end of the file, not necessary.